### ISSUE TITLE
[QUEST] Remove jQuery

### ISSUE THREAD

**Issue by simonihmig** [emberjs/ember.js#17476]
This is an issue to track the progress on implementing the [RFC386](https://github.com/emberjs/rfcs/pull/386) to remove jQuery by default.

#### [ember.js](https://github.com/emberjs/ember.js)

- [x] add a (universal, non-silenceable) deprecation warning to `Ember.$()`
🔒@simonihmig, PR: https://github.com/emberjs/ember.js/pull/17489
- [x] add a deprecation warning to `this.$()` in an `Ember.Component`
🔒@simonihmig, PR: https://github.com/emberjs/ember.js/pull/17488
- [x] make sure user sees deprecation warning of jQuery integration itself, i.e. neither the `jquery-integration` optional feature flag is set to false nor `@ember/jquery` has been installed.
🔒@simonihmig, PR: https://github.com/ember-cli/ember-cli/pull/8399
- [x] wrap all jQuery related code for svelte
🔒@simonihmig, PR: https://github.com/emberjs/ember.js/pull/17525

#### [@ember/jquery](https://github.com/emberjs/ember-jquery)

- [x] Add support for `Component.$()`
🔒@simonihmig, WIP PR: https://github.com/emberjs/ember-jquery/pull/27
- [ ] Add jQuery-based EventDispatcher replacement 
🔒@simonihmig, WIP PR: https://github.com/emberjs/ember-jquery/pull/31
- [x] Make `import $ from 'jquery';` work
🔒@simonihmig, PR: https://github.com/emberjs/ember-jquery/pull/33

#### [ember-cli](https://github.com/ember-cli/ember-cli)

- [ ] Update app blueprint, stage 1: Update `@ember/jquery` to latest version
PR: https://github.com/ember-cli/ember-cli/pull/8396
- [ ] Update app blueprint, stage 2: remove jQuery
🔒@cyriller, PR: https://github.com/ember-cli/ember-cli/pull/8394

#### [@ember/test-helpers](https://github.com/emberjs/ember-test-helpers)

- [x] Add a deprecation warning to `this.$()` in component integration tests, based on `setupRenderingTest()`
🔒@simonihmig, PR: https://github.com/emberjs/ember-test-helpers/pull/543

#### [ember-cli-babel](https://github.com/babel/ember-cli-babel)

- [x] Do not transpile `import $ from 'jquery';` to `Ember.$` if the right version of `@ember/jquery` is present
🔒@simonihmig, PR: https://github.com/babel/ember-cli-babel/pull/261

#### [ember-data](https://github.com/emberjs/data)

- [ ] Make ember-data work with ember-fetch
🔒@tchak PR: https://github.com/emberjs/data/pull/5386

#### [ember-fetch](https://github.com/ember-cli/ember-fetch)

- [ ] Provide a replacement API for jQuery's `prefilter`

#### Website

- [x] Add a [deprecation guide](https://emberjs.github.io/rfcs/0386-remove-jquery.html#how-we-teach-this)
🔒@simonihmig PR: https://github.com/ember-learn/deprecation-app/pull/255

#### [ember.js](https://github.com/emberjs/ember.js)

- [ ] remove deprecated functions
- [ ] remove the jQuery specific code paths in the EventDispatcher

## Collaboration

If someone wants to start work on a specific item, please post this here so I can update the list.

**Comment by simonihmig** [comment]
@chriskrycho you mentioned in Discord that you could take over the ember-fetch/prefilter part. Is this still the case, can I assign you to this task? 👆

**Comment by chriskrycho** [comment]
Depends on when we want it done by. I'm starting a new job two weeks from today (*very* much still in Ember land), so I'm in wrapping-up-all-the-things mode now and will be in starting-all-the-things mode for the first couple weeks after that. I'm happy to do it, but given all fo that it'll probably be the second or third week of February before it's viable.

**Comment by mactenm** [comment]
its not my what i do best  if you dont have time must admid it isnt my
strong side . but allow me to thank you for that help you could give

man. 14. jan. 2019 kl. 18.25 skrev Chris Krycho <notifications@github.com>:

**Comment by bobisjan** [comment]
Thanks for this issue 👏!

I have a question if FastBoot should be included in this quest with deprecating and removing

- `najax` from [fastboot](
https://github.com/ember-fastboot/fastboot),
- `ajax` initializer from [ember-cli-fastboot](
https://github.com/ember-fastboot/ember-cli-fastboot)?

**Comment by rwjblue** [comment]
@simonihmig - We need to add svelte support to the deprecations added in #17489 and #17488. The basic process is:

* add an entry in `packages/@ember/deprecated-features/index.ts`
* import that constant and use it to wrap all deprecated code paths in an `if` (I think this actually means _all_ code that would be using the preexisting jQueryDisabled flag _also_ gets the new guard)

Some recent examples:

* https://github.com/emberjs/ember.js/pull/16744
* https://github.com/emberjs/ember.js/pull/16956

**Comment by dcyriller** [comment]
@simonihmig you may want to add a lock on the item: "Update app blueprint"

**Comment by simonihmig** [comment]
@dcyriller cool, done! 👍 

I am just thinking, switching the app blueprints to remove jQuery by default is certainly blocked by the outstanding ember-data issue. If that is not resolved by the time Ember 3.9 lands, we should probably do an intermediate step and update the blueprints to just bump the `@ember/jquery` version to 0.6.0, so an `ember new` app does not see any deprecations. And then switch to no jQuery mode as soon as ember-data is ready. @rwjblue agree?

**Comment by simonihmig** [comment]
Closing this in favor of https://github.com/emberjs/rfc-tracking/issues/3!
